src/shared/pages/welcomePage/script.js

```diff
@@ -20,41 +20,47 @@ export async function checkPrivateBrowsing() {
   const alwaysPrivateCheckbox = document.getElementById(
     "always-private-checkbox",
   );
+  const alwaysPrivateSlider = document.getElementById("always-private-slider");
+
   const currentExternalBrowser = await getExternalBrowser();
   if (currentExternalBrowser === "Firefox Private Browsing") {
     alwaysPrivateCheckbox.checked = true;
+    alwaysPrivateSlider.ariaChecked = true;
   } else {
     alwaysPrivateCheckbox.checked = false;
+    alwaysPrivateSlider.ariaChecked = false;
   }
 
-  // On change, update the default launch mode
-  alwaysPrivateCheckbox.addEventListener("change", async () => {
+  browser.storage.sync.onChanged.addListener((changes) => {
+    if (changes.currentExternalBrowser !== undefined) {
+      if (
+        changes.currentExternalBrowser.newValue === "Firefox Private Browsing"
+      ) {
+        alwaysPrivateCheckbox.checked = true;
+        alwaysPrivateSlider.ariaChecked = true;
+      } else {
+        alwaysPrivateCheckbox.checked = false;
+        alwaysPrivateSlider.ariaChecked = false;
+      }
+    }
+  });
+
+  async function handleAlwaysPrivateSwitchClick() {
     let from = await getExternalBrowser();
     await handleChangeDefaultLaunchContextMenuClick();
     let to = await getExternalBrowser();
 
-    if (to === "Firefox Private Browsing") {
-      alwaysPrivateCheckbox.checked = true;
-    } else {
-      alwaysPrivateCheckbox.checked = false;
-    }
-
     settingEvent.currentBrowser.record({
       from,
       to,
       source: "always_private_checkbox",
     });
-  });
+  }
 
-  browser.storage.sync.onChanged.addListener((changes) => {
-    if (changes.currentExternalBrowser !== undefined) {
-      if (
-        changes.currentExternalBrowser.newValue === "Firefox Private Browsing"
-      ) {
-        alwaysPrivateCheckbox.checked = true;
-      } else {
-        alwaysPrivateCheckbox.checked = false;
-      }
+  alwaysPrivateSlider.addEventListener("click", handleAlwaysPrivateSwitchClick);
```

Comment:
The "slider" is an element inside a `<label>` that wraps an `<input>` element. A click on content within `<label>` triggers interaction with the `<input>` field. E.g. with text input field it would focus the field, in this case the checkbox toggle state would change.

Adding a click handler to the slider element inside the slider results in any click to be processed twice.